File: Source/Alternet.UI/Controls/AuiManager/AuiToolbarArt.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Alternet.UI
{
    internal class AuiToolbarArt : DisposableObject, IAuiToolbarArt
    {
        internal AuiToolbarArt(IntPtr handle, bool disposeHandle)
            : base(handle, disposeHandle)
        {
        }

        public void SetFlags(uint flags)
        {
            Native.AuiToolBarArt.SetFlags(Handle, flags);
        }

        public uint GetFlags()
        {
            return Native.AuiToolBarArt.GetFlags(Handle);
        }

        public void SetTextOrientation(AuiToolbarTextOrientation orientation)
        {
            Native.AuiToolBarArt.SetTextOrientation(Handle, (int)orientation);
        }

        public AuiToolbarTextOrientation GetTextOrientation()
        {
            return (AuiToolbarTextOrientation)Native.AuiToolBarArt.GetTextOrientation(Handle);
        }

        // Note that these functions work with the size in DIPs, not physical pixels.
        public int GetElementSize(int elementId)
        {
            return Native.AuiToolBarArt.GetElementSize(Handle, elementId);
        }

        public void SetElementSize(int elementId, int size)
        {
            Native.AuiToolBarArt.SetElementSize(Handle, elementId, size);
        }

        // Provide opportunity for subclasses to recalculate colors
        public void UpdateColorsFromSystem()
        {
            Native.AuiToolBarArt.UpdateColorsFromSystem(Handle);
        }
    }
}

File: Old/Removed 2024/RemovedNativeApi/RemovedFromPal/AuiToolBarArt.cpp
#include "AuiToolBarArt.h"

namespace Alternet::UI
{
	AuiToolBarArt::AuiToolBarArt()
	{

	}

	AuiToolBarArt::~AuiToolBarArt()
	{
	}
	
	void AuiToolBarArt::SetFlags(void* handle, uint32_t flags)
	{
		((wxAuiToolBarArt*)handle)->SetFlags(flags);
	}

	uint32_t AuiToolBarArt::GetFlags(void* handle)
	{
		return ((wxAuiToolBarArt*)handle)->GetFlags();
	}

	void AuiToolBarArt::SetTextOrientation(void* handle, int orientation)
	{
		((wxAuiToolBarArt*)handle)->SetTextOrientation(orientation);
	}

	int AuiToolBarArt::GetTextOrientation(void* handle)
	{
		return ((wxAuiToolBarArt*)handle)->GetTextOrientation();
	}

	int AuiToolBarArt::GetElementSize(void* handle, int elementId)
	{
		return ((wxAuiToolBarArt*)handle)->GetElementSize(elementId);
	}

	void AuiToolBarArt::SetElementSize(void* handle, int elementId, int size)
	{
		((wxAuiToolBarArt*)handle)->SetElementSize(elementId, size);
	}

	void AuiToolBarArt::UpdateColorsFromSystem(void* handle)
	{
		((wxAuiToolBarArt*)handle)->UpdateColoursFromSystem();
	}
}
